lib/services/project-service.ts

```diff
@@ -99,21 +106,32 @@ export class ProjectService implements IProjectService {
 		return null;
 	}
 
-	private async extractTemplate(projectDir: string, realTemplatePath: string): Promise<void> {
+	private async extractTemplate(projectDir: string, realTemplatePath: string, templateVersion: string): Promise<void> {
 		this.$fs.ensureDirectoryExists(projectDir);
 
-		const appDestinationPath = this.$projectData.getAppDirectoryPath(projectDir);
-		this.$fs.createDirectory(appDestinationPath);
+		this.$logger.trace(`Template version is ${templateVersion}`);
+		let destinationDir = "";
+		switch (templateVersion) {
+			case constants.TemplateVersions.v2:
```

Comment:
Because we always defatult to v1 [here](https://github.com/NativeScript/nativescript-cli/pull/3542/files#diff-123e2a8a4e8daddf9ac8a60b8c79546dR65), maybe will be better that the `default` case is v2 and above.
```Typescript
case constants.TemplateVersions.v1:
    const appDestinationPath = this.$projectData.getAppDirectoryPath(projectDir);
    this.$fs.createDirectory(appDestinationPath);
    destinationDir = appDestinationPath;
    break;
default:
   destinationDir = projectDir;
```